mesh.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "indexbuffer.h"
+#include "texture.h"
+#include "vertexarray.h"
+#include "vertexbuffer.h"
+
+#include <glm/glm.hpp>
+
+#include <memory>
+
+class Mesh
+{
+protected:
+    VertexArray mVA {};
+    std::unique_ptr<VertexBuffer>mVB { nullptr };
+    std::unique_ptr<IndexBuffer>mIB { nullptr };
```

Comment:
Curious if you have any thoughts here @cjappl. These classes need their verticies known on instantation in order to render properly. But as a interface this Mesh class has no verticies known, so it cant really allocate these on the stack (at least with the current way these classes are setup). My best work around was making them heap allocated. I know one of our old problems was that everything in a lot of our classes were pointers, when they didn't need to be.

I think it is possible to have a late instantation, something like:
```
VertexBuffer vb;
vb.SetBufferData(positions.data(), positions.size() * sizeof(float))
```
But that is kind of a similar pattern to our old string class
```
// Create one
splString text;
// Now make it useful
text.Copy("Hello World");
```
Which I was never a fan of either

Reply:
The last one is so convoluted I ended up writing up a big thing and then realizing I don't understand the implications well enough to suggest it :) homework for me I guess

Reply:
Yeah I decided to go with your first suggestions of an empty instantation and assigning them later on. I had to implement the copy constructors to do it. If you could look at those and make sure they look good that would be great. I also implemented the move constructors just to practice a bit doing it. Thanks!